File: FitToFit/FitToFit.Services/Database/Akcije.cs
using System;
using System.Collections.Generic;

namespace FitToFit.Services.Database;

public partial class Akcije
{
    public int AkcijaId { get; set; }

    public string Naziv { get; set; } = null!;

    public DateTime DatumPocetka { get; set; }

    public DateTime DatumZavrsetka { get; set; }

    public int Iznos { get; set; }

    public virtual ICollection<AkcijeTreninzi> AkcijeTreninzis { get; set; } = new List<AkcijeTreninzi>();
}

File: FitToFit/UI/fittofit_mobile/lib/models/akcije.dart
import 'package:fittofit_mobile/models/akcijeTreninzi.dart';
import 'package:json_annotation/json_annotation.dart';
part 'akcije.g.dart';

@JsonSerializable()
class Akcije {
  int akcijaId;
  String naziv;
  DateTime datumPocetka;
  DateTime datumZavrsetka;
  int iznos;
  String? stateMachine;
  List<AkcijeTreninzi>? akcijeTreninzis = [];

  Akcije(
      {required this.akcijaId,
      required this.naziv,
      required this.datumPocetka,
      required this.datumZavrsetka,
      required this.iznos,
      this.stateMachine,
      this.akcijeTreninzis});

  factory Akcije.fromJson(Map<String, dynamic> json) => _$AkcijeFromJson(json);

  Map<String, dynamic> toJson() => _$AkcijeToJson(this);
}
